OpenRA.Game/Settings.cs

```diff
@@ -300,6 +302,52 @@ public void Save()
 			root.WriteToFile(settingsFile);
 		}
 
+		public static string SanitizedName(string dirty)
+		{
+			// reserved characters for MiniYAML and JSON
+			var disallowedChars = new char[] { '#', '@', ':', '\n', '\t', '[', ']', '{', '}', '"', '`' };
+
+			var clean = dirty;
+			if (string.IsNullOrEmpty(dirty))
+				clean = "";
+			else
+			{
+				foreach (var disallowedChar in disallowedChars)
+					clean = clean.Replace(disallowedChar.ToString(), string.Empty);
```

Comment:
I'd suggest replacing with wither '-' or '_', because otherwise this can return an empty string, which leads us to my comment above.